pyvisa/resources/messagebased.py

```diff
@@ -475,8 +479,9 @@ def read_binary_values(self, datatype='f', is_big_endian=False,
                                                              is_big_endian)
             expected_length = offset + data_length
 
-        if self._read_termination is not None:
-            expected_length += len(self._read_termination)
+        if expect_termination:
```

Comment:
This can use a simple and and does not need to be split on two lines.

Reply:
ok